plugins/node/opentelemetry-instrumentation-memcached/src/instrumentation.ts

```diff
@@ -0,0 +1,183 @@
+    );
+  }
+
+  setConfig(config: InstrumentationConfig = Instrumentation.DEFAULT_CONFIG) {
+    this._config = Object.assign({}, Instrumentation.DEFAULT_CONFIG, config);
+  }
+
+  init() {
+    return [
+      new InstrumentationNodeModuleDefinition<typeof Memcached>(
+        'memcached',
+        ['>=2.2'],
+        (moduleExports, moduleVersion) => {
+          api.diag.debug(
+            `Patching ${Instrumentation.COMPONENT}@${moduleVersion}`
+          );
+          this.ensureWrapped(
+            moduleVersion,
+            moduleExports.prototype,
+            'command',
+            this.wrapCommand.bind(this, moduleVersion)
+          );
+          return moduleExports;
+        },
+        (moduleExports, moduleVersion) => {
+          api.diag.debug(
+            `Unpatching ${Instrumentation.COMPONENT}@${moduleVersion}`
+          );
+          if (moduleExports === undefined) return;
+          // `command` is documented API missing from the types
+          this._unwrap(moduleExports.prototype, 'command' as keyof Memcached);
+        }
+      ),
+    ];
+  }
+
+  wrapCommand(
+    moduleVersion: undefined | string,
+    original: (
+      queryCompiler: () => Memcached.CommandData,
+      server?: string
+    ) => any
+  ) {
+    const instrumentation = this;
+    return function (
+      this: Memcached,
+      queryCompiler: () => Memcached.CommandData,
+      server?: string
+    ) {
+      if (typeof queryCompiler !== 'function') {
+        return original.apply(this, arguments as any);
+      }
+      // The name will be overwritten later
+      const span = instrumentation.tracer.startSpan('<type> <key>', {
+        kind: api.SpanKind.CLIENT,
+        attributes: {
+          'memcached.version': moduleVersion,
+          ...Instrumentation.COMMON_ATTRIBUTES,
+        },
+      });
+      const context = api.trace.setSpan(api.context.active(), span);
+
+      return api.context.with(
+        context,
+        original,
+        this,
+        instrumentation.wrapQueryCompiler.call(
+          instrumentation,
+          queryCompiler,
+          this,
+          server,
+          context,
+          span
+        ),
+        server
+      );
+    };
+  }
+
+  wrapQueryCompiler(
+    original: () => Memcached.CommandData,
+    client: Memcached,
+    server: undefined | string,
+    context: api.Context,
+    span: api.Span
+  ) {
+    const instrumentation = this;
+    return function (this: Memcached) {
+      const query = original.apply(this, arguments as any);
+      const callback = query.callback;
+
+      span.updateName(`${query.type} ${query.key}`);
+      span.setAttributes({
+        'db.memcached.key': query.key,
+        'db.memcached.lifetime': query.lifetime,
+        [SemanticAttributes.DB_OPERATION]: query.type,
+        'db.statement': (instrumentation._config as InstrumentationConfig)
+          .collectCommand
+          ? query.command
```

Comment:
I don't this API that well, and based on your tests it looks like the answer is 'no', but I have to ask: can this leak values, which might be sensitive information?  It looks like it's just operation and/or key names, but I want to make sure.

Reply:
By default, it does not leak values.

However, if the configuration explicitly has `collectCommand` set to `true`, it does when setting values for example. Thoughts on that?

Reply:
The behavior is fine and the existence of the config boolean is also fine; I might add a stronger disclaimer in the documentation - "Caution: setting `collectCommand=true` exposes values in command strings, which may contain sensitive information." or similar.